src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,38 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
+
+    public BasePage(WebDriver givenDriver){
+        driver = givenDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions = new Actions(driver);
+        }
+
+    public void click(By locator) {
+        wait.until(ExpectedConditions.elementToBeClickable(locator));
+
+
+
+    }
+
+//            public WebElement findElement(By locator){
+//        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
+//    }
+//    public void performDoubleClick (WebElement locator){
+//        wait.until(ExpectedConditions.elementToBeClickable(locator));
+//        actions.doubleClick(locator).perform();
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.